tests/python_package_test/test_basic.py

```diff
@@ -997,3 +997,39 @@ def test_equal_datasets_from_one_and_several_matrices_w_different_layouts(rng, t
     lgb.Dataset(mat)._dump_text(one_path)
 
     assert filecmp.cmp(one_path, several_path)
+
+
+def test_string_serialized_params_retrieval(rng):
+    # Random train data
+    train_x = rng.random((500, 20))
+    train_y = rng.integers(0, 1, 500)
+    train_data = lgb.Dataset(train_x, train_y)
+
+    # Parameters
+    params = {
+        "boosting": "gbdt",
+        "objective": "binary",
+        "metric": ["auc"],
+        "num_leaves": 31,
+        "learning_rate": 0.05,
+        "feature_fraction": 0.9,
+        "bagging_fraction": 0.8,
+        "bagging_freq": 5,
+        "verbosity": -100,
+    }
+
+    # train a model and serialize it to a string in memory
+    model = lgb.train(params, train_data)
```

Comment:
```suggestion
        "num_leaves": 7
        "learning_rate": 0.05,
        "feature_fraction": 0.9,
        "bagging_fraction": 0.8,
        "bagging_freq": 5,
        "verbosity": -100,
    }

    # train a model and serialize it to a string in memory
    model = lgb.train(params, train_data, num_boost_round=2)
```

Let's please train an even smaller, cheaper model. The model's statistical performance and total number of leaves isn't really important here, since this is about how parameters survive a serialization-deserialization loop.